proto/catalog.proto

```diff
@@ -63,8 +63,17 @@ message StreamSourceInfo {
   optional string key_message_name = 11;
   plan_common.ExternalTableDesc external_table = 12;
   // Whether the stream source is a cdc source streaming job.
-  // We need this field to differentiate the cdc source job until we fully implement risingwavelabs/rfcs#72.
+  // We need this field to differentiate the cdc source job, because CDC source is a singleton fragment, while Kafka source is not.
+  // If we have other singleton source jobs in the future, we may consider deprecating this field and add a new field `is_distributed`.
   bool cdc_source_job = 13;
+  // Whether the stream source is a shared source (it has a streaming job).
+  // This is related with [RFC: Reusable Source Executor](https://github.com/risingwavelabs/rfcs/pull/72).
+  // Currently, the following sources have streaming jobs:
+  // - Direct CDC sources (mysql & postgresql). Backwards compat note: For old CDC job it's `None`; For new CDC job it's `Some(true)`.
+  // - MQ sources (Kafka, Pulsar, Kinesis, etc.)
+  //
+  // **Should also test `cdc_source_job` for backwards compatibility. Use `is_shared_compatible()` instead of this field directly.**
```

Comment:
Looks error-prone to me. I would rather just reuse `cdc_source_job` even without renaming.

Reply:
😄 Sure that's acceptable to me

Reply:
Had better reuse **and** rename 😄